v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -85,31 +149,96 @@ function isDiffableError(
   );
 }
 
-function improveNodeAssertTitle(title: string, error: Error): string {
+function getErrorDiff(error: Error): string | undefined {
   if (!isDiffableError(error)) {
-    return title;
+    return undefined;
   }
 
-  if (!title.includes("AssertionError [ERR_ASSERTION]: ")) {
-    return title;
+  if ("showDiff" in error && error.showDiff === false) {
+    return undefined;
   }
 
-  const match = title.match(/^AssertionError \[ERR_ASSERTION\]\: (.*)\:/);
+  return getDiff(error.expected, error.actual) ?? undefined;
+}
+
+/**
+ * This function parses a single stack trace line and returns the parsed
+ * reference or undefined.
+ *
+ * Parsable stack trace lines are of the form:
+ * - at <context> (<location>:<lineNumber>:<columnNumber>)
+ * - at <context> (<location>:<lineNumber>)
+ * - at <context> (<location>)
+ * - at <location>:<lineNumber>:<columnNumber>
+ * - at <location>:<lineNumber>
+ * - at <location>
+ *
+ * @param line
+ * @returns
+ */
+export function parseStackLine(line: string): StackReference | undefined {
+  const regex = /^at (?:(.+?) \()?(?:([^\(].+?)(?::(\d+))?(?::(\d+))?)\)?$/;
```

Comment:
What do you think about doing something like this?

```ts
const regex = new RegExp(
  '^at ' +                         // Start with "at " 
  '(?:' +                           // Non-capturing group for optional context:
  '(.+?) \\(' +                     //   - Captures the context (lazy match) followed by " (" 
  ')?' +                            // End optional context group
  '(?:' +                           // Non-capturing group for location, line number, and column:
  '([^\\(].+?)' +                   //   - Captures the location (anything except a "(" at the start, lazy match)
  '(?::(\\d+))?' +                  //   - Optional: captures the line number prefixed by ":"
  '(?::(\\d+))?' +                  //   - Optional: captures the column number prefixed by ":"
  ')?' +                            // End non-capturing group for location and numbers
  '\\)?$',                          // Optional closing ")" at the end and end of the string
  ''
);
```

Note that it was generated by chatgpt, so it may be incorrect 😅

Reply:
That's a great idea, I'll add it - I'll double check if LLM was correct here ;)